servers/physics/space_sw.cpp

```diff
@@ -624,9 +624,6 @@ int SpaceSW::test_body_ray_separation(BodySW *p_body, const Transform &p_transfo
 
 					ShapeSW *against_shape = col_obj->get_shape(shape_idx);
 					if (CollisionSolverSW::solve_static(body_shape, body_shape_xform, against_shape, col_obj->get_transform() * col_obj->get_shape_transform(shape_idx), cbkres, cbkptr, NULL, p_margin)) {
-						if (cbk.amount > 0) {
```

Comment:
Sounds like this was a bug and maybe the intent was likely to check `ckbres.amount` (i.e. the callback result from the call on the previous line).

Reply:
CC @reduz

Reply:
`Space2DSW::test_body_ray_separation` has a similar issue BTW.

Reply:
This is intended, the value is modified by the callback passed to the function above

Reply:
Some more info from IRC:
```
<reduz> Akien: calling solve() will trigger a callback that will modify the amount
14:50 <reduz> Akien: so if this was triggered by an analyzer its probably a bug in the analyzer
14:51 <reduz> Akien: removing this would cause a bug
14:51 <Akien> reduz: so `cbk` is modified via `cbkptr` I guess?
14:51 <reduz> yes, this should be reported to the static analyzer
```

Reply:
What was the report from the static analyzer BTW? I can't find a reference to this line in your four issues.

Reply:
```
godot/servers/physics/space_sw.cpp	627	err	V547 Expression 'cbk.amount > 0' is always false.
godot/servers/physics/space_sw.cpp	646	err	V654 The condition 'k < cbk.amount' of loop is always false.
godot/servers/physics_2d/space_2d_sw.cpp	619	err	V547 Expression 'cbk.amount > 0' is always false.
godot/servers/physics_2d/space_2d_sw.cpp	639	err	V654 The condition 'k < cbk.amount' of loop is always false.
```